src/test/java/net/syscon/elite/aop/connectionproxy/OracleConnectionAspectTest.java

```diff
@@ -0,0 +1,81 @@
+package net.syscon.elite.aop.connectionproxy;
+
+import net.syscon.elite.security.AuthSource;
+import net.syscon.elite.security.AuthenticationFacade;
+import oracle.jdbc.driver.OracleConnection;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Properties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+
+class OracleConnectionAspectTest {
+
+    private final AuthenticationFacade authenticationFacade = mock(AuthenticationFacade.class);
+    private final RoleConfigurer roleConfigurer = mock(RoleConfigurer.class);
+    private final String defaultSchema = "some default schema";
+    private final Connection pooledConnection = mock(Connection.class);
+    private final PreparedStatement pooledPreparedStatement = mock(PreparedStatement.class);
+    private final OracleConnection proxyConnection = mock(OracleConnection.class);
+    private final PreparedStatement proxyPreparedStatement = mock(PreparedStatement.class);
+
+    private final OracleConnectionAspect connectionAspect = new OracleConnectionAspect(authenticationFacade, roleConfigurer, defaultSchema);
+
+    @Test
+    void openProxySessionIfIdentifiedAuthentication_nomisUser_opensProxyConnection() throws SQLException {
+        configureMocks(AuthSource.NOMIS);
+
+        connectionAspect.openProxySessionIfIdentifiedAuthentication(pooledConnection);
+
+        ArgumentCaptor<Properties> propertiesCaptor = ArgumentCaptor.forClass(Properties.class);
+        verify(proxyConnection).openProxySession(eq(OracleConnection.PROXYTYPE_USER_NAME), propertiesCaptor.capture());
+        assertThat(propertiesCaptor.getValue().get((OracleConnection.PROXY_USER_NAME))).isEqualTo("some user name");
+    }
+
+    @Test
+    void openProxySessionIfIdentifiedAuthentication_nomisUser_setSchemaAndContext() throws SQLException {
+        configureMocks(AuthSource.NOMIS);
+
+        connectionAspect.openProxySessionIfIdentifiedAuthentication(pooledConnection);
+
+        ArgumentCaptor<String> sqlCaptor = ArgumentCaptor.forClass(String.class);
+        verify(pooledConnection, times(2)).prepareStatement(sqlCaptor.capture());
+        assertThat(sqlCaptor.getAllValues().get(0)).contains("ALTER SESSION SET CURRENT_SCHEMA");
```

Comment:
Perhaps use `assertThat(.....).containsExactly(....)`instead, same for other tests.

Reply:
I do this on purpose to prevent tests from becoming brittle.  For example in this case I'm just testing that the current schema is being set but I don't really care what it is being set to.